Node Status Monitor/NodeJS/node_monitor.js

```diff
@@ -0,0 +1,197 @@
+function fetch_Data(token) {
+    return new Promise((resolve, reject) => {
+        log.info("-------------------- fetching Node Details --------------------")
+        fetch(nodes_detail_url, {
+            headers: {
+                'accept': 'application/json',
+                'content-length': '0',
+                'authorization': `Bearer ${token}`
+            }
+        })
+            .then(res => res.json())
+            .then(json => {
+                // if object has propeery Message ,display Error, else Process Data
+                if (json.hasOwnProperty('Message')) {
+                    log.error(`************************* ${json.Message} *************************`);
+                    reject(json.Message)
+                } else {
+                    resolve(json.items)
+                }
+
+            }).catch(err => {
+                reject(err)
+            }
+            );
+    });
+}
+
+/*function to process Node Details */
+function process_node_status(new_data) {
+    log.info("-------------------- processing Node Details --------------------")
+    let new_node_data = new_data.map(item => {
+        return {
+            id: item.id,
+            node_name: item.name,
+            status: item.status.name,
+            network_type: item.network_type.name
+        }
+    });
+
+    return new_node_data;
+}
+
+/*function to write Node Details */
+function write_data(data) {
+    log.info("-------------------- writing  Node Details --------------------")
+    let parsed_data = JSON.stringify(data);
+
+    fs.writeFile(old_data_file, parsed_data, 'utf8', function () {
+        log.info("-------------------- writing Old Node Details --------------------")
+    });
+
+    fs.writeFile(new_data_file, parsed_data, 'utf8', function () {
+        log.info("-------------------- writing New Node Details --------------------")
+    });
+}
+
+/*function to read old Node Details */
+function read_old_data() {
+    return new Promise((resolve, reject) => {
+        if (fs.existsSync(old_data_file)) {
+            // Do something
+            log.info("-------------------- Reading Old Node Details --------------------")
+            fs.readFile(old_data_file, 'utf8', function readFileCallback(err, data) {
+                if (err) {
+                    reject(err);
+                } else {
+                    if (data) {
+                        let old_data = JSON.parse(data); //now it an object
+                        resolve(old_data)
+                    } else {
+                        resolve(data)
+                    }
+
+                }
+            });
+        } else {
+            resolve()
+        }
+    });
+
+}
+
+/*function to compare  Node Details */
+function compare_data(old_data, new_data) {
+    log.info("-------------------- Comparing  Node Details --------------------")
+    let temp = [];
+    var unique_status_change_node_result = old_data.filter(function (obj) {
```

Comment:
```suggestion
    var updated_nodes = old_data.filter(function (obj) {
```
and continue down with rename.

Reply:
Done!